package-kuring/Sources/UIKit/BotUI/BotView.swift

```diff
@@ -0,0 +1,168 @@
+            if isSendPopupVisible {
+                sendPopup
+                    .transition(.opacity)
+                    .zIndex(1)
+            }
+        }
+        .onAppear {
+            store.send(.onAppear)
+        }
+        .navigationBarBackButtonHidden()
+    }
+    
+    private var headerView: some View {
+        HStack {
+            backButton
+            Spacer()
+            titleText
+            Spacer()
+            infoButton
+        }
+        .padding(.horizontal, 18)
+    }
+    
+    private var backButton: some View {
+        Button {
+            dismiss()
+        } label: {
+            Image(systemName: "chevron.backward")
+                .padding()
+                .frame(width: 30, height: 20)
+                .foregroundStyle(Color.Kuring.gray400)
+        }
+    }
+    
+    private var titleText: some View {
+        Text("쿠링봇")
+            .padding()
+            .font(.system(size: 18, weight: .semibold))
+    }
+    
+    private var infoButton: some View {
+        Button {
+            isPopoverVisible.toggle()
+        } label: {
+            Image("icon_info_circle", bundle: Bundle.bots)
+        }
+        .popover(isPresented: $isPopoverVisible, arrowEdge: .top) {
+            popoverContent
+        }
+    }
+    
+    private var popoverContent: some View {
+        VStack(spacing: 10) {
+            Text("• 쿠링봇은 2024년 6월 이후의 공지\n  사항 내용을 기준으로 답변할 수 있\n  어요.")
+            Text("• 테스트 기간인 관계로 한 달에 2회\n  까지만 질문 가능해요.")
+        }
+        .lineSpacing(5)
+        .font(.system(size: 15, weight: .medium))
+        .padding(20)
+        .presentationBackground(Color.Kuring.gray100)
+        .presentationCompactAdaptation(.popover)
+    }
+    
+    @ViewBuilder
+    private var chatView: some View {
+        if !store.state.chatHistory.isEmpty {
+            ChatView(store: self.store)
+        } else {
+            ChatEmptyView()
+        }
+    }
+    
+    private var inputView: some View {
+        HStack(alignment: .bottom, spacing: 12) {
+            TextField("질문을 입력해주세요", text: $tempInputText.limit(to: 300), axis: .vertical)
+                .lineLimit(5)
+                .focused($isInputFocused)
+                .padding(.horizontal)
+                .padding(.vertical, 12)
+                .overlay(RoundedRectangle(cornerRadius: 20).strokeBorder(Color.Kuring.gray200, style: StrokeStyle(lineWidth: 1.0)))
+            
+            sendButton
+        }
+        .padding(.horizontal, 20)
+        .disabled(store.chatHistory.count >= 4)
+    }
+    
+    private var sendButton: some View {
+        Button {
+            isInputFocused = false
+            isSendPopupVisible = true
+        } label: {
+            Image(systemName: "arrow.up.circle.fill")
+                .resizable()
+                .foregroundStyle(Color.Kuring.gray400)
+                .scaledToFit()
+                .frame(width: 40, height: 40)
+        }
+        .disabled(store.chatHistory.count >= 4)
+    }
```

Comment:
TCA 구조로 코드를 작성하실거라면 아래와 같은 모습이 될 수 있게 시도해보시는 걸 추천드립니다.

그리고 TCA 는 필수가 아니지만 TCA 로 작업할거라면 유지보수를 위해 TCA 사용 부분은 확실하게 TCA 로 가는 게 좋습니다.

```suggestion
    private var sendButton: some View {
        Button {
            store.send(.sendButtonTapped)
        } label: {
            Image(systemName: "arrow.up.circle.fill")
                .resizable()
                .foregroundStyle(Color.Kuring.gray400)
                .scaledToFit()
                .frame(width: 40, height: 40)
        }
        .disabled(store.isSendDisabled)
    }
```

```swift
case .sendButtonTapped:
    state.isInputFocused = false
    state.isSendPopupVisible = true
```

Reply:
넵 ! 같이 수정해서 올리겠습니다 ! 감사합니다